### PR TITLE
doc/rados: edit troubleshooting-pg (2 of x)

### PR DESCRIPTION
**Pull request by zdover23**
Edit doc/rados/troubleshooting/troubleshooting-pg.rst (1 of x).

Follows https://github.com/ceph/ceph/pull/54052.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/rados/troubleshooting/troubleshooting-pg.rst [hunk 1]
```diff
@@ -349,77 +349,106 @@ may confuse applications that expect the object to exist.
 Homeless Placement Groups
 =========================
 
-It is possible for all OSDs that had copies of a given placement groups to fail.
-If that's the case, that subset of the object store is unavailable, and the
-monitor will receive no status updates for those placement groups.  To detect
-this situation, the monitor marks any placement group whose primary OSD has
-failed as ``stale``.  For example::
+It is possible that ever OSD that had copies of a given placement group could
```
Comment (anthonyeleven) [inline doc/rados/troubleshooting/troubleshooting-pg.rst:352]: Every

Comment (anthonyeleven) [reply doc/rados/troubleshooting/troubleshooting-pg.rst:352]: Suggest recasting the paragraph in present tense, can fail , if this happens …

File: doc/rados/troubleshooting/troubleshooting-pg.rst [hunk 2]
```diff
@@ -349,77 +349,106 @@ may confuse applications that expect the object to exist.
...
 ================
 
-If your cluster is up, but some OSDs are down and you cannot write data,
-check to ensure that you have the minimum number of OSDs running for the
-placement group. If you don't have the minimum number of OSDs running,
-Ceph will not allow you to write data because there is no guarantee
-that Ceph can replicate your data. See ``osd_pool_default_min_size``
-in the `Pool, PG and CRUSH Config Reference`_ for details.
+If the cluster is up, but some OSDs are down and you cannot write data, make
+sure that you have the minimum number of OSDs running for the placement group.
```
Comment (anthonyeleven) [inline doc/rados/troubleshooting/troubleshooting-pg.rst:408]: We might mean pool here vs pg

File: doc/rados/troubleshooting/troubleshooting-pg.rst [hunk 3]
```diff
@@ -473,82 +502,99 @@ objects. If an object named ``foo`` in PG ``0.6`` is truncated, we will have::
...
 
-You can repair the inconsistent placement group by executing::
+     * ``data_digest_mismatch_info``: the digest stored in the ``object-info``
+       is not ``0xffffffff``, which is calculated from the shard read from
+       ``OSD.2``
+     * ``size_mismatch_info``: the size stored in the ``object-info`` is
+       different from the size read from ``OSD.2``. The latter is ``0``.
 
-	ceph pg repair {placement-group-ID}
+To repair the inconsistent placement group, run a command of the following
```
Comment (anthonyeleven) [inline doc/rados/troubleshooting/troubleshooting-pg.rst:534]: We should encourage examining dmesg, smartctl etc for drive errors before blithely attempting a repair.

File: doc/rados/troubleshooting/troubleshooting-pg.rst [hunk 4]
```diff
@@ -473,82 +502,99 @@ objects. If an object named ``foo`` in PG ``0.6`` is truncated, we will have::
...
+   does not work in every case. For example, it might be the case that the
+   stored data digest is missing, which means that the calculated digest is
+   ignored when Ceph chooses the authoritative copies. Be aware of this, and
+   use the above command with caution.
 
-If ``read_error`` is listed in the ``errors`` attribute of a shard, the
-inconsistency is likely due to disk errors. You might want to check your disk
-used by that OSD.
+If ``read_error`` is listed in a shard's ``errors`` attribute, the
+inconsistency is likely due to physical storage errors. In cases like this,
```
Comment (anthonyeleven) [inline doc/rados/troubleshooting/troubleshooting-pg.rst:550]: Like I said ;). Maybe move this before the discussion of issuing a repair.

File: doc/rados/troubleshooting/troubleshooting-pg.rst [hunk 5]
```diff
@@ -473,82 +502,99 @@ objects. If an object named ``foo`` in PG ``0.6`` is truncated, we will have::
...
 
      [2,1,6,0,5,8,2147483647,7,4]
 
 Not enough OSDs
 ---------------
 
-If the Ceph cluster only has 8 OSDs and the erasure coded pool needs
-9, that is what it will show. You can either create another erasure
-coded pool that requires less OSDs::
+If the Ceph cluster has only eight OSDs and the erasure coded pool needs nine
```
Comment (anthonyeleven) [inline doc/rados/troubleshooting/troubleshooting-pg.rst:571]: s/and the/and a/

File: doc/rados/troubleshooting/troubleshooting-pg.rst [hunk 6]
```diff
@@ -566,36 +612,43 @@ the rule::
...
-  greater than the default.
+* using a handmade CRUSH rule that tries more times to find a good mapping.
+  This can be done by setting ``set_choose_tries`` to a value greater than the
+  default.
 
-You should first verify the problem with ``crushtool`` after
-extracting the crushmap from the cluster so your experiments do not
-modify the Ceph cluster and only work on a local files::
+First, verify the problem by using  ``crushtool`` after extracting the crushmap
+from the cluster. This ensure that your experiments do not modify the Ceph
```
Comment (anthonyeleven) [inline doc/rados/troubleshooting/troubleshooting-pg.rst:643]: Ensures

File: doc/rados/troubleshooting/troubleshooting-pg.rst [hunk 7]
```diff
@@ -566,36 +612,43 @@ the rule::
...
 
-* adding more OSDs to the cluster (that does not require the erasure
-  coded pool to be modified, it will become clean automatically)
+* adding more OSDs to the cluster (this does not require the erasure coded pool
+  to be modified, because it will become clean automatically)
 
-* use a handmade CRUSH rule that tries more times to find a good
-  mapping. This can be done by setting ``set_choose_tries`` to a value
-  greater than the default.
+* using a handmade CRUSH rule that tries more times to find a good mapping.
```
Comment (anthonyeleven) [inline doc/rados/troubleshooting/troubleshooting-pg.rst:638]: This can be modified for an existing CRUSh rule.

**Review by anthonyeleven — APPROVED** [review]

File: doc/rados/troubleshooting/troubleshooting-pg.rst (hunk 1 above)
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-pg.rst:352]: Good catch, and I accept the suggest to change the paragraph's tense.

Accepted.

File: doc/rados/troubleshooting/troubleshooting-pg.rst (hunk 2 above)
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-pg.rst:408]: I will clear this up in a few hours and return in a future PR if this needs alteration. I do understand the issue here.

File: doc/rados/troubleshooting/troubleshooting-pg.rst (hunk 3 above)
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-pg.rst:534]: Accepted.

File: doc/rados/troubleshooting/troubleshooting-pg.rst (hunk 4 above)
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-pg.rst:550]: Accepted with thanks.

File: doc/rados/troubleshooting/troubleshooting-pg.rst (hunk 5 above)
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-pg.rst:571]: I went with s/and the/and an/. Good catch.

File: doc/rados/troubleshooting/troubleshooting-pg.rst (hunk 6 above)
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-pg.rst:643]: Good catch. Thank you.

File: doc/rados/troubleshooting/troubleshooting-pg.rst (hunk 7 above)
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-pg.rst:638]: Accepted.

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/54114 - Reef backport
https://github.com/ceph/ceph/pull/54115 - Quincy backport

File: doc/rados/troubleshooting/troubleshooting-pg.rst (hunk 2 above)
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-pg.rst:408]: https://github.com/ceph/ceph/pull/54218

**Pull request merged**